tests/Aspire.Dashboard.Tests/Integration/DashboardClientAuthTests.cs

```diff
@@ -0,0 +1,177 @@
+        Assert.NotNull(call.RequestHeaders);
+        Assert.Null(call.RequestHeaders.Get(ApiKeyHeaderName));
+    }
+
+    [Fact]
+    public async Task ConnectsToResourceService_ApiKey()
+    {
+        await using var server = await CreateResourceServiceServerAsync();
+        await using var client = await CreateDashboardClientAsync(server.Url, authMode: ResourceClientAuthMode.ApiKey, configureOptions: options => options.ResourceServiceClient.ApiKey = "TestApiKey!");
+
+        var call = server.Calls.ApplicationInformationCalls.Single();
+
+        Assert.NotNull(call.Request);
+        Assert.NotNull(call.RequestHeaders);
+        Assert.Equal("TestApiKey!", call.RequestHeaders.GetValue(ApiKeyHeaderName));
+    }
+
+    private static async Task<ResourceServiceServer> CreateResourceServiceServerAsync(Action<TestCalls>? configureCalls = null)
+    {
+        var serverAppBuilder = WebApplication.CreateSlimBuilder();
+
+        TestCalls testCalls = new();
+
+        configureCalls?.Invoke(testCalls);
+
+        serverAppBuilder.Services.AddGrpc(options => options.EnableDetailedErrors = true);
+        serverAppBuilder.Services.AddSingleton(testCalls);
+        serverAppBuilder.WebHost.ConfigureKestrel(ConfigureKestrel);
+
+        var serverApp = serverAppBuilder.Build();
+
+        serverApp.MapGrpcService<MockDashboardService>();
+        serverApp.MapGet("/", () => "Communication with gRPC endpoints must be made through a gRPC client.");
+
+        await serverApp.StartAsync();
+
+        return new(serverApp, testCalls);
+
+        static void ConfigureKestrel(KestrelServerOptions kestrelOptions)
+        {
+            // Listen on a random port.
+            kestrelOptions.Listen(IPAddress.Loopback, port: 0, ConfigureListen);
+
+            static void ConfigureListen(ListenOptions options)
+            {
+                // Force HTTP/2 for gRPC, so that it works over non-TLS connections
+                // which cannot negotiate between HTTP/1.1 and HTTP/2.
+                options.Protocols = HttpProtocols.Http2;
+            }
+        }
+    }
+
+    private static async Task<DashboardClient> CreateDashboardClientAsync(
+        string serverAddress,
+        ResourceClientAuthMode authMode = ResourceClientAuthMode.Unsecured,
+        Action<DashboardOptions>? configureOptions = null)
+    {
+        var options = new DashboardOptions
+        {
+            ResourceServiceClient =
+            {
+                AuthMode = authMode,
+                Url = serverAddress
+            }
+        };
+
+        configureOptions?.Invoke(options);
+
+        options.ResourceServiceClient.TryParseOptions(out _);
+
+        DashboardClient client = new(
+            loggerFactory: NullLoggerFactory.Instance,
+            configuration: new ConfigurationManager(),
+            dashboardOptions: Options.Create(options));
+
+        var iClient = (IDashboardClient)client;
+
+        await iClient.WhenConnected;
```

Comment:
Can WhenConnected stop being explicitly implemented? I don't see the value and it makes using and testing the property a little harder.